x-pack/apm-server/main.go

```diff
@@ -118,6 +119,33 @@ func newTailSamplingProcessor(args beater.ServerParams) (*sampling.Processor, er
 	}
 	readWriter := getStorage(badgerDB)
 
+	if tailSamplingConfig.StorageLimitAuto {
+		const (
+			// fallbackDefault is the default TBS storage limit if disk space detection fails.
+			fallbackDefault = 3 << 30
```

Comment:
[to reviewer] It is still going to be an arbitrary number, but do we want to increase this to something like 5GB or 10GB?

Reply:
With the discussed solution, WDYT about changing the default to the % of total disk size, e.g. 70%.